system_setup/server/controllers/configure.py

```diff
@@ -143,29 +143,35 @@ async def _install_check_lang_support_packages(self, lang, env) -> bool:
             return False
 
         packages = [pkg for pkg in cp.stdout.strip().split(' ') if pkg]
-        if len(packages) == 0:
```

Comment:
Note that the removal of this means that the non-dry-run case does touch the cache with an open/commit set of calls.  I doubt that's a problem but wanted to confirm that this is intentional or at least benign.

Reply:
Thanks for pointing that out.
That could lead to unexpected effects if there were other packages marked to install previously by other routines. Whether that's feasible or not under the context of system_setup I couldn't figure out, honestly. But I made sure the cache is only touched if the packages list is not empty. Safer.